test/testdrive/temporary.td

```diff
@@ -67,6 +67,19 @@ catalog item 'foo' already exists
 
 > CREATE OR REPLACE TEMPORARY MATERIALIZED VIEW foo AS SELECT * FROM v
 
+> CREATE TEMPORARY TABLE temp_t (a int, b text NOT NULL)
+
+! CREATE TEMP TABLE temp_t (a int, b text NOT NULL)
+catalog item 'temp_t' already exists
+
+> INSERT INTO temp_t VALUES (1, 'testing')
+
```

Comment:
```ALTER TABLE RENAME TO``` causes an index out of bounds panic

```
pstoev=> create temporary table t1 (f1 integer);
CREATE TABLE
pstoev=> alter table t1 rename to t2;
server closed the connection unexpectedly

2021-02-03T08:38:37.077042Z ERROR panic: <unnamed>: index out of bounds: the len is 2 but the index is 2
   0: materialized::handle_panic
             at src/materialized/src/bin/materialized/main.rs:569:65
   1: core::ops::function::Fn::call
             at /rustc/7eac88abb2e57e752f3302f02be5f3ce3d7adfb4/library/core/src/ops/function.rs:70:5
   2: std::panicking::rust_panic_with_hook
             at /rustc/7eac88abb2e57e752f3302f02be5f3ce3d7adfb4/library/std/src/panicking.rs:581:17
   3: std::panicking::begin_panic_handler::{{closure}}
             at /rustc/7eac88abb2e57e752f3302f02be5f3ce3d7adfb4/library/std/src/panicking.rs:484:9
   4: std::sys_common::backtrace::__rust_end_short_backtrace
             at /rustc/7eac88abb2e57e752f3302f02be5f3ce3d7adfb4/library/std/src/sys_common/backtrace.rs:153:18
   5: rust_begin_unwind
             at /rustc/7eac88abb2e57e752f3302f02be5f3ce3d7adfb4/library/std/src/panicking.rs:483:5
   6: core::panicking::panic_fmt
             at /rustc/7eac88abb2e57e752f3302f02be5f3ce3d7adfb4/library/core/src/panicking.rs:85:14
   7: core::panicking::panic_bounds_check
             at /rustc/7eac88abb2e57e752f3302f02be5f3ce3d7adfb4/library/core/src/panicking.rs:62:5
   8: <usize as core::slice::index::SliceIndex<[T]>>::index_mut
             at /rustc/7eac88abb2e57e752f3302f02be5f3ce3d7adfb4/library/core/src/slice/index.rs:261:14
      core::slice::index::<impl core::ops::index::IndexMut<I> for [T]>::index_mut
             at /rustc/7eac88abb2e57e752f3302f02be5f3ce3d7adfb4/library/core/src/slice/index.rs:26:9
      <alloc::vec::Vec<T> as core::ops::index::IndexMut<I>>::index_mut
             at /rustc/7eac88abb2e57e752f3302f02be5f3ce3d7adfb4/library/alloc/src/vec.rs:1994:9
      sql::ast::transform::create_stmt_rename
             at src/sql/src/ast/transform.rs:35:13
   9: coord::catalog::CatalogItem::rename_item_refs::{{closure}}
             at src/coord/src/catalog.rs:329:17
  10: coord::catalog::CatalogItem::rename_item_refs
             at src/coord/src/catalog.rs:339:32
  11: coord::catalog::Catalog::transact
             at src/coord/src/catalog.rs:1285:32
  12: coord::coord::Coordinator<C>::catalog_transact::{{closure}}
             at src/coord/src/coord.rs:2789:22
      <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
             at /rustc/7eac88abb2e57e752f3302f02be5f3ce3d7adfb4/library/core/src/future/mod.rs:80:19
  13: coord::coord::Coordinator<C>::sequence_alter_item_rename::{{closure}}
             at src/coord/src/coord.rs:2752:15
      <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
             at /rustc/7eac88abb2e57e752f3302f02be5f3ce3d7adfb4/library/core/src/future/mod.rs:80:19
      coord::coord::Coordinator<C>::sequence_plan::{{closure}}
             at src/coord/src/coord.rs:1660:17
  14: <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
             at /rustc/7eac88abb2e57e752f3302f02be5f3ce3d7adfb4/library/core/src/future/mod.rs:80:19
      coord::coord::Coordinator<C>::message_statement_ready::{{closure}}
             at src/coord/src/coord.rs:500:17
      <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
             at /rustc/7eac88abb2e57e752f3302f02be5f3ce3d7adfb4/library/core/src/future/mod.rs:80:19
      coord::coord::Coordinator<C>::serve::{{closure}}
             at src/coord/src/coord.rs:402:21
      <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
             at /rustc/7eac88abb2e57e752f3302f02be5f3ce3d7adfb4/library/core/src/future/mod.rs:80:19
  15: tokio::park::thread::CachedParkThread::block_on::{{closure}}
             at /home/pstoev/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.1.0/src/park/thread.rs:263:54
      tokio::coop::with_budget::{{closure}}
             at /home/pstoev/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.1.0/src/coop.rs:121:9
      std::thread::local::LocalKey<T>::try_with
             at /rustc/7eac88abb2e57e752f3302f02be5f3ce3d7adfb4/library/std/src/thread/local.rs:272:16
      std::thread::local::LocalKey<T>::with
             at /rustc/7eac88abb2e57e752f3302f02be5f3ce3d7adfb4/library/std/src/thread/local.rs:248:9
      tokio::coop::with_budget
             at /home/pstoev/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.1.0/src/coop.rs:114:5
      tokio::coop::budget
             at /home/pstoev/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.1.0/src/coop.rs:98:5
      tokio::park::thread::CachedParkThread::block_on
             at /home/pstoev/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.1.0/src/park/thread.rs:263:31
  16: tokio::runtime::enter::Enter::block_on
             at /home/pstoev/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.1.0/src/runtime/enter.rs:151:13
      tokio::runtime::thread_pool::ThreadPool::block_on
             at /home/pstoev/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.1.0/src/runtime/thread_pool/mod.rs:71:9
  17: tokio::runtime::Runtime::block_on
             at /home/pstoev/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.1.0/src/runtime/mod.rs:452:43
  18: coord::coord::serve::{{closure}}::{{closure}}
             at src/coord/src/coord.rs:3421:31
      std::sys_common::backtrace::__rust_begin_short_backtrace
             at /rustc/7eac88abb2e57e752f3302f02be5f3ce3d7adfb4/library/std/src/sys_common/backtrace.rs:137:18
  19: std::thread::Builder::spawn_unchecked::{{closure}}::{{closure}}
             at /rustc/7eac88abb2e57e752f3302f02be5f3ce3d7adfb4/library/std/src/thread/mod.rs:464:17
      <std::panic::AssertUnwindSafe<F> as core::ops::function::FnOnce<()>>::call_once
             at /rustc/7eac88abb2e57e752f3302f02be5f3ce3d7adfb4/library/std/src/panic.rs:308:9
      std::panicking::try::do_call
             at /rustc/7eac88abb2e57e752f3302f02be5f3ce3d7adfb4/library/std/src/panicking.rs:381:40
      std::panicking::try
             at /rustc/7eac88abb2e57e752f3302f02be5f3ce3d7adfb4/library/std/src/panicking.rs:345:19
      std::panic::catch_unwind
             at /rustc/7eac88abb2e57e752f3302f02be5f3ce3d7adfb4/library/std/src/panic.rs:382:14
      std::thread::Builder::spawn_unchecked::{{closure}}
             at /rustc/7eac88abb2e57e752f3302f02be5f3ce3d7adfb4/library/std/src/thread/mod.rs:463:30
      core::ops::function::FnOnce::call_once{{vtable.shim}}
             at /rustc/7eac88abb2e57e752f3302f02be5f3ce3d7adfb4/library/core/src/ops/function.rs:227:5
  20: <alloc::boxed::Box<F> as core::ops::function::FnOnce<A>>::call_once
             at /rustc/7eac88abb2e57e752f3302f02be5f3ce3d7adfb4/library/alloc/src/boxed.rs:1042:9
      <alloc::boxed::Box<F> as core::ops::function::FnOnce<A>>::call_once
             at /rustc/7eac88abb2e57e752f3302f02be5f3ce3d7adfb4/library/alloc/src/boxed.rs:1042:9
      std::sys::unix::thread::Thread::new::thread_start
             at /rustc/7eac88abb2e57e752f3302f02be5f3ce3d7adfb4/library/std/src/sys/unix/thread.rs:87:17
  21: start_thread
  22: clone
    
materialized encountered an internal error and crashed.
```

Reply:
Great catch! This is also an issue for views. For the curious, [the bug is here](https://github.com/MaterializeInc/materialize/blob/main/src/sql/src/ast/transform.rs#L37).

Reply:
Fixed this locally! A more satisfying fix will come with https://github.com/MaterializeInc/materialize/issues/5591